packages/test/test-version-utils/src/compatConfig.ts

```diff
@@ -130,6 +131,49 @@ const genLTSConfig = (compatVersion: number | string): CompatConfig[] => {
 	];
 };
 
+const genBackCompatConfig = (compatVersion: number): CompatConfig[] => {
+	return [
+		{
+			name: `compat back N${compatVersion} - older loader`,
+			kind: CompatKind.Loader,
+			compatVersion,
+			loader: compatVersion,
+		},
+		{
+			name: `compat back N${compatVersion} - older loader + older driver`,
+			kind: CompatKind.LoaderDriver,
+			compatVersion,
+			driver: compatVersion,
+			loader: compatVersion,
+		},
+	];
+};
+
+const genFullBackCompatConfig = (): CompatConfig[] => {
+	const _configList: CompatConfig[] = [];
+	// This will need to be updated once we move beyond 2.0.0-internal.x.y.z
+	let semverInternal: string | undefined;
+	if (pkgVersion.startsWith("2.0.0-internal.")) {
+		semverInternal = pkgVersion.split("internal.")[1];
+	}
+
+	if (pkgVersion.startsWith("2.0.0-dev.")) {
+		semverInternal = pkgVersion.split("dev.")[1];
+	}
+
+	assert(semverInternal !== undefined, "Unexpected pkg version");
+	// This is to make pipeline runs work
+	semverInternal = semverInternal.split(".").slice(0, 3).join(".");
+
+	const semverVal = new semver.SemVer(semverInternal);
+	const num = semverVal.major;
+	// This makes the assumption N and N-1 scenarios are already fully tested thus skipping 0 and -1.
+	for (let i = 2; i < num; i++) {
+		_configList.push(...genBackCompatConfig(0 - i));
```

Comment:
This is a v1, I decided it would be cheaper to get this out so it's definitely just a first step. It would be nice to generate a matrix based on production.

Reply:
We want to go back to LTS so we already do N and N-1 and LTS scenarios. So all we need to do is N-2 => LTS+1 which is 2.0.0-internal.1.4.x

Reply:
My understanding is that we say we support the greatest minor. This PR was made with that assumption.

Reply:
This loop goes as far back as 2.0.0.internal.1.y.z.
The idea is to generate all the versions from -2 -> - (major - 1) the current major version (i.e 2.0.0-internal.9.y.z would be -8)
This means as the number of majors increase the number of versions we support - this may be updated in the future..